src/utils/map-controls.js

```diff
@@ -98,9 +106,9 @@ export default class MapControls {
 
   /* Callback util */
   // formats map state and invokes callback function
-  updateViewport(newMapState, extraState = {}) {
+  updateViewport(newMapState, extraProps = {}, extraState = {}) {
```

Comment:
Does the code get more clean if you put last two props into single object ?

something like :
```
updateViewport(newMapState, {extraProps = NO_TRANSITION_PROPS, extraState = {}})

when calling updateViewport you can set only extraProps or extraState or both.
```

Reply:
Good point - it probably does. But React convention is to keep the props and state objects separate in parameters, I think we should follow that as much as we can.